src/REstate/Engine/TransitionFailedPreconditionException.cs

```diff
@@ -1,48 +1,21 @@
 ﻿using System;
-using REstate.Engine.Connectors;
-using REstate.Schematics;
 
 namespace REstate.Engine
 {
-    public class TransitionFailedPreconditionException<TState, TInput, TPayload>
```

Comment:
Changing this to non-generic since this needs to move across network boundaries, we need to keep exceptions minimal.